### PR TITLE
Graph Team Tag CRUD - CSharp

### PR DESCRIPTION
**Pull request by HunaidHanfee-MSFT**

### CODE REVIEW

File: samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  "icons": {
+    "color": "color.png",
+    "outline": "outline.png"
+  },
+  "name": {
+    "short": "Manage Team Tags",
+    "full": "Manage Team Tags"
+  },
+  "description": {
+    "short": "This example can be used to manage Teams team tags.",
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json:22]: This example is used to manage Teams team tags.",

File: samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "full": "This example can be used to manage Teams team tags. You can create, update, add or remove members. All of Graph CRUD operations related to tags can be performed within this sample."
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json:23]: This example is used to manage Teams team tags.",

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/README.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/README.md:3]: Do we need this file?

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/public/index.html [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Meeting Stage View Sample"
+    />
+    <title>Meetings Stage View</title>
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/public/index.html:11]: This is not meeting stage sample :)

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/src/components/view-or-edit-tag.jsx [hunk 5]
```diff
@@ -0,0 +1,282 @@
+// <copyright file="view-or-edittag.tsx" company="Microsoft Corporation">
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/src/components/view-or-edit-tag.jsx:1]: we can rename this file as view-edit-tag.jsx

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+        }
+
+        /// <summary>
+        /// Create team tag.
+        /// </summary>
+        /// <param name="teamId">Id of team.</param>
+        /// <param name="teamTag">Details of the tag to be created.</param>
+        /// <returns>If success return 201 status code, otherwise 500 status code</returns>
+        [HttpPost("{teamId}")]
+        public async Task<IActionResult> CreateTeamTagAsync([FromRoute] string teamId, [FromBody]TeamTagUpdateDto teamTag)
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:33]: Add proper spacing here

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs [hunk 7]
```diff
@@ -0,0 +1,145 @@
...
+            }
+        }
+
+        /// <summary>
+        /// List all the tags for the specified team.
+        /// </summary>
+        /// <param name="teamId">Id of the team.</param>
+        /// <returns>If success return list of tags, otherwise 500 status code</returns>
+        [HttpGet("{teamId}/list")]
+        public async Task<IActionResult> ListTeamTagAsync([FromRoute]string teamId)
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:72]: Add proper spacing here

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+        }
+
+        /// <summary>
+        /// Updates the tag.
+        /// </summary>
+        /// <param name="teamId">Id of team.</param>
+        /// <param name="teamTag">Updated details of the tag.</param>
+        /// <returns>If success return 204 status code, otherwise 500 status code</returns>
+        [HttpPatch("{teamId}/update")]
+        public async Task<IActionResult> UpdateTeamTagAsync([FromRoute] string teamId, [FromBody]TeamTagUpdateDto teamTag)
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:92]: Add proper spacing here

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Helper/GraphHelper.cs [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+
+            var teamworkTagUpdated = await this.graphBetaClient.Teams[teamId].Tags[teamTag.Id].Request().UpdateAsync(teamworkTag);
+            if (teamworkTagUpdated != null)
+            {
+                await AddRemoveTagMembersAsync(teamTag, teamId);
+            }
+        }
+
+        /// <summary>
+        /// Updates the member of tag.
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/GraphTeamsTag/Helper/GraphHelper.cs:133]: Add or remove

File: samples/graph-teams-tag/csharp/README.md [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+- office
+- office-365
+languages:
+- csharp
+extensions:
+contentType: samples
+createdDate: "24-06-2022 00:02:15"
+---
+
+# This is a sample application shows the usage of Graph CRUD operations related to team tags.
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/README.md:15]: This is a sample application that shows the usage of Graph CRUD operations related to team tags.

File: samples/graph-teams-tag/csharp/README.md [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+- **Upload** the `manifest.zip` to Teams (in the Apps view click "Upload a custom app")
+
+## Features of this sample
+
+1. User can see list of tags created for the current team.
+![Manage Tag Dashboard](GraphTeamsTag/Images/Dashboard.png)
+2. User can view/edit the existing team tags.
+![View/Edit Tags](GraphTeamsTag/Images/ViewOrEditTag.png)
+3. User can create new team tags.
+![Create new Tag](GraphTeamsTag/Images/CreateTagTaskModule.png)
```
Comment (ChetanSharma-msft) [inline samples/graph-teams-tag/csharp/README.md:95]: Please check if images are showing properly in readme.

**Review by ChetanSharma-msft — CHANGES_REQUESTED** [review]
Reviewed and added my comments

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/README.md (hunk 3 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/README.md:3]: Okay we can remove

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/public/index.html (hunk 4 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/public/index.html:11]: Done 😄

File: samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/src/components/view-or-edit-tag.jsx (hunk 5 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/ClientApp/src/components/view-or-edit-tag.jsx:1]: Done

File: samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs (hunk 7 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/Controllers/TeamTagController.cs:72]: Done

File: samples/graph-teams-tag/csharp/README.md (hunk 11 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/README.md:95]: Yes it is

File: samples/graph-teams-tag/csharp/README.md (hunk 10 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/README.md:15]: Done

File: samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json (hunk 1 above)
Comment (HunaidHanfee-MSFT) [reply samples/graph-teams-tag/csharp/GraphTeamsTag/AppManifest/manifest.json:22]: Done

**Review by ChetanSharma-msft — APPROVED** [review]
Changes looks fine, approving the PR.

**Review by Wajeed-msft — APPROVED** [review]

**Pull request merged**